### ISSUE TITLE
[question] Managing large binary 3rd party dependencies

### ISSUE THREAD

**Issue by walterj-adsk** [conan-io/conan#7844]
We are building a conan package based on a large (closed-source) 3rd party dependency that exceeds several GB in size.  I wanted to use conan because I can bundle the dependency as a conan package and our CI/CD pipelines will automatically fetch the dependency.  Otherwise, I have to manually add the dependency to our build machine.  

I have a simple recipe : 
```
from conans import ConanFile
class Party3rdConan(ConanFile):
    name = "Party3rd"
    version = "1.0.1"
    settings = "os", "arch"
    description = "Large Dependency"
    url = ""
    license = "Commercial"
    author = "ACME"
    topics = None

    def package(self):
        self.copy("*")

    def package_info(self):
        pass
```

I run export-pkg and it properly creates the conan package.  However, if I try to upload the package to our artifactory server it fails to complete because of the size of the dependency.  I think conan is crashing or over-consuming on the OS side which is causing it to crash.  The binary is about 6GB and compressed version is 3.5GB.  I'm wondering if there is a way to tell upload to split the binary package into parts and re-assemble those parts.  This would break up the binary size and allow parallization of the upload.

**Comment by memsharded** [comment]
Hi @walterj-adsk 

Certainly, this is a large package :)

We have been reported issues in the past with quite large packages (+1-2Gb) in the transfers, bit more often download than upload. In theory, they are loaded from disk and uploaded incrementally, like streaming, to avoid depleting resources. There is also resumable downloads in place for unreliable networks. There is nothing there for explicitly splitting those large files into smaller chunks.

Sometimes, these transfers issues happen only in some platforms, Windows, for example, while it can work in others. Have you tried the same but from other OS? Also, the python architecture itself might help, and 64bits version (the default Python in Windows is 32bits!) might behave a bit better. If you can try the same upload in other machine, that could help to triage if it is a platform issue.

It would be nice to have some performance/benchmark e2e testing for this, something that we could share and run in different places to help debugging these issues, lets add something.

**Comment by walterj-adsk** [comment]
I'm running Windows with Python 64bit.  I have not tried another OS but I'm exploring another option now that is outlined on the conan documentation:  https://docs.conan.io/en/latest/devtools/create_installer_packages.html

Basically the build step would just download the binaries from another source.  Not ideal but I think it will allow me to work around the problem for now.

**Comment by memsharded** [comment]
> Basically the build step would just download the binaries from another source. Not ideal but I think it will allow me to work around the problem for now.

Yes, that might be a pragmatic workaround. Though the download of huge things might (very occasionally, but we have seen it before) fail too. Keep us updated about this workaround in case it has issues too. Thanks!

**Comment by AbrilRBS** [comment]
Closing as solved as we didn't hear back since then. The upload/download reliability has improved in Conan 2, but we still have seem some users have different issues for large packages. If anyone faces issues like this, please feel free to create a new issue, thanks!
